crates/sui-indexer-alt/migrations/2024-12-02-185822_add_coin_balance_owner_kind/up.sql

```diff
@@ -0,0 +1,14 @@
+ALTER TABLE sum_coin_balances
+ADD COLUMN coin_owner_kind SMALLINT NOT NULL DEFAULT 1;
+
+DROP INDEX sum_coin_balances_owner_type;
+CREATE INDEX sum_coin_balances_owner_type
```

Comment:
I'm thinking of some ways to make this easier to land without causing the indexer to hang when it starts, and I think I have an idea:

- We can get postgres to drop and create these indices concurrently, which will mean they won't hang other things that are going on at the same time (this will allow us to run the migrations proactively while the indexer is running), 
- and then turn them into `IF EXISTS`/`IF NOT EXISTS` style statements so that they become idempotent, (which means after we've run it proactively, when the migration runs, it knows not to drop and regenerate the indices),
- and finally, we need to give the new index a new name so that `IF EXISTS`/`IF NOT EXISTS` would recognise it as different from the old index.

The statements for creating and dropping the indices would look like this then:

```suggestion
DROP INDEX CONCURRENTLY IF EXISTS sum_coin_balances_owner_type;
CREATE INDEX CONCURRENTLY IF NOT EXISTS sum_coin_balances_owner_kind_and_id;
```

But you will also need to split it up over separate migration scripts (because each script is its own transaction and postgres won't like it if you combine non-blocking concurrent operations with other work in a transaction). I've had to do this once before in #19543 so that can serve as a template (there's an extra TOML file that needs to go in each migration folder to tell diesel to not batch up migrations into a transaction as well).

(ditto for the wal tables below).

Reply:
I did this, let me know if it looks correct. when I run `generate_schema.sh` locally it fails with an error 
```
2024-12-05 10:37:44.702 EST [78369] ERROR:  DROP INDEX CONCURRENTLY cannot run inside a transaction block
```

But I have the `metadata.toml` files there as you suggested - is anything else missing?

Reply:
The issue is that the migrations need to be split up even further -- each `DROP INDEX` and `CREATE INDEX` needs to be in their own script (because they can't mingle with each other either).

Reply:
Thanks, done